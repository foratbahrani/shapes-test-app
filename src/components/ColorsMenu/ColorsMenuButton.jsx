import React from 'react';
import Shape from '../Shape/Shape';

const ColorsMenuButton = ({ color }) => {
	return <Shape key={color} color={color} shape='square' size='sm' />;
};

export default ColorsMenuButton;
