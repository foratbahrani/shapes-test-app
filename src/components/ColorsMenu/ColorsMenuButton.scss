.colorsMenuButton {
	appearance: none;
	border: none;
	background-color: transparent;
	cursor: pointer;
	padding: 0;
	margin: 0 8px;
	height: 30px;
}

.colorsMenuButton:focus {
	outline: 0;
}
