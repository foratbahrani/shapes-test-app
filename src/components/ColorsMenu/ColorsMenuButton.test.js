import { shallow } from 'enzyme';
import React from 'react';
import { ColorsMenuButton } from './ColorsMenuButton';

let wrapper;
let toggleFilteredColor = undefined;

beforeEach(() => {
	const color = 'red';
	const filteredColors = ['red'];
	toggleFilteredColor = jest.fn().mockReturnValue('');
	wrapper = shallow(
		<ColorsMenuButton
			color={color}
			filteredColors={filteredColors}
			toggleFilteredColor={toggleFilteredColor}
		/>
	);
});

it('matches snapshot', () => {
	expect(wrapper).toMatchSnapshot();
});

it('calls toggle function', () => {
	expect.assertions(2);
	wrapper.find('[id="button"]').simulate('click');
	expect(toggleFilteredColor.mock.calls.length).toBe(1);
	wrapper.find('[id="button"]').simulate('click');
	expect(toggleFilteredColor.mock.calls.length).toBe(2);
});
