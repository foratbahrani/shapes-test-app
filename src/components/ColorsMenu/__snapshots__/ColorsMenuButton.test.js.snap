// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="colorsMenuButton shrink"
  id="button"
  onClick={[Function]}
>
  <Shape
    color="red"
    isActive={true}
    key="red"
    shape="square"
    size="sm"
  />
</button>
`;
